server/src/main/java/org/candlepin/model/AbstractHibernateObject.java

```diff
@@ -33,12 +33,17 @@
 
 /**
  * Abstract class for hibernate entities
+ *
+ * @param <T>
+ *  Entity type extending this class; should be the name of the subclass
  */
 @MappedSuperclass
 @XmlType(name = "CandlepinObject")
 @JsonIgnoreProperties(ignoreUnknown = true)
 @JsonFilter("DefaultFilter")
-public abstract class AbstractHibernateObject implements Persisted, Serializable {
+public abstract class AbstractHibernateObject<T extends AbstractHibernateObject>
```

Comment:
Okay, I think it would make a little more sense to look like:

CandlepinDTO
```java
public CandlepinDTO populate(T source) {
```

TimestampedCandlepinDTO
```java
public TimestampedCandlepinDTO populate(T source) {
```
and then in OwnerDTO
```java
public OwnerDTO populate(OwnerDTO source) {
```

That compiles for me when I implement it everywhere and has the bonus of removing lines like

```java
        // We're safeguarded here by both the generic definition above, and the runtime-level check
        // in CandlepinDTO's populate method.
        CertificateSerialDTO src = (CertificateSerialDTO) source;
```